portfolio/src/main/webapp/script.js

```diff
@@ -32,21 +32,37 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+ 
+/**
+ * Fetches and loads header and footer content for each page.
+ */
+function loadHeadFoot() {
+  // Fetch & load header
+  fetch("templates/header.html")
+    .then(response => response.text())
+    .then(text => {
+      let header = document.createElement("header");
+      header.innerHTML = text;
+      document.querySelector("body").before(header);
+    });
```

Comment:
Looks like you do the same thing here and in fetching the footer.  Perhaps you can abstract out into a helper function and parameterize using 'header' vs. 'footer' where needed.

Reply:
Great idea. I created a function for loading html into the DOM and used that for the header, footer, and greeting.

Reply:
Great refactoring and use of comments in your code!